### ISSUE TITLE
optimum threshold detection

### ISSUE THREAD

**Issue by Ashwith-ram** [rafaelpadilla/Object-Detection-Metrics#148]
Hi @rafaelpadilla, I have tried plotting the PR curve, But I want to choose an optimum threshold for class confidence for predictions based on the curve. How do I do that?

**Comment by rafaelpadilla** [comment]
Hi @Ashwith-ram ,

It all depends what is "optimum" for you. 

Let me explain better: The PR curve is plotted considering different confidence levels showing a trade-off between precision and recall. 

Take a look at [this illustrated example](https://github.com/rafaelpadilla/Object-Detection-Metrics#an-ilustrated-example). Look at the [table](https://github.com/rafaelpadilla/Object-Detection-Metrics/blob/master/aux_images/table_2_v2.png): If your application does care much about missing objects, but it requires that all detected objects are correct, you would pick confidence 98%, as it gives you Precision=1 and a low Recall=0.0666. Nevertheless, it will retrieve only one object (see the column Acc TP).

But, on the other hand, if your application needs to retrieve as many objects as possible, even if they are incorrect, you would choose a confidence that gives you the highest recall. But, in this example, you would be retrieving more false positives. This analysis can also be made on the curve, but the curve does not show the confidence levels. That's why looking at the table to choose the confidence level is better.

In summary, the optimal confidence threshold is all about the needs of your scenario, and you can choose it looking at the table.

I hope it was clear. Best regards

**Comment by Ashwith-ram** [comment]
Hi @rafaelpadilla ,

Yes, I completely agree with you, I wanted a comparison between precision, recall and f1. Nevertheless, I made a small addition to your code so that it would return precision, recall values for all confidence values. 
Thanks

**Comment by yashsandansing** [comment]
Hey @Ashwith-ram , do you mind sharing the code that returns precision, recall values at different confidence values?
